editor/modes/visual-editor/block.js

```diff
@@ -234,6 +237,63 @@ class VisualEditorBlock extends Component {
 		this.props.onSelect();
 	}
 
+	onKeyDown( event ) {
+		const { keyCode } = event;
+
+		this.handleArrowKey( event );
+
+		if ( keyCode === UP || keyCode === LEFT || keyCode === DOWN || keyCode === RIGHT ) {
+			const selection = window.getSelection();
+			this.lastRange = selection.rangeCount ? selection.getRangeAt( 0 ) : null;
+		}
+	}
+
+	onKeyUp( event ) {
+		this.removeOrDeselect( event );
+		this.handleArrowKey( event );
+	}
+
+	handleArrowKey( event ) {
+		const { keyCode, target } = event;
+		const moveUp = ( keyCode === UP || keyCode === LEFT );
+		const moveDown = ( keyCode === DOWN || keyCode === RIGHT );
+		const selectors = [
+			'*[contenteditable="true"]',
+			'*[tabindex]',
+			'textarea',
+			'input',
+		].join( ',' );
```

Comment:
Yeah, I'm not sure if we want to match all tabbables here, or only textarea-like ones?